src/main/java/com/epam/ta/reportportal/ws/rabbit/AsyncReportingListener.java

```diff
@@ -0,0 +1,310 @@
+import com.epam.ta.reportportal.util.ProjectExtractor;
+import com.epam.ta.reportportal.ws.converter.builders.AttachmentBuilder;
+import com.epam.ta.reportportal.ws.converter.builders.LogBuilder;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import com.epam.ta.reportportal.ws.model.FinishExecutionRQ;
+import com.epam.ta.reportportal.ws.model.FinishTestItemRQ;
+import com.epam.ta.reportportal.ws.model.StartTestItemRQ;
+import com.epam.ta.reportportal.ws.model.launch.StartLaunchRQ;
+import com.epam.ta.reportportal.ws.model.log.SaveLogRQ;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.AmqpRejectAndDontRequeueException;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageListener;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.epam.ta.reportportal.commons.EntityUtils.normalizeId;
+import static com.epam.ta.reportportal.core.configs.rabbit.ReportingConfiguration.DEAD_LETTER_MAX_RETRY;
+import static com.epam.ta.reportportal.core.configs.rabbit.ReportingConfiguration.EXCHANGE_REPORTING_RETRY;
+import static com.epam.ta.reportportal.core.configs.rabbit.ReportingConfiguration.QUEUE_DLQ;
+
+/**
+ * @author Konstantin Antipin
+ */
+public class AsyncReportingListener implements MessageListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AsyncReportingListener.class);
+
+    @Autowired
+    MessageConverter messageConverter;
+
+    @Autowired
+    @Qualifier("rabbitTemplate")
+    AmqpTemplate amqpTemplate;
+
+
+    @Autowired
+    private StartLaunchHandler startLaunchHandler;
+
+    @Autowired
+    private FinishLaunchHandler finishLaunchHandler;
+
+    @Autowired
+    private StartTestItemHandler startTestItemHandler;
+
+    @Autowired
+    private FinishTestItemHandler finishTestItemHandler;
+
+
+    @Autowired
+    private DatabaseUserDetailsService userDetailsService;
+
+
+    @Autowired
+    private LogRepository logRepository;
+
+    @Autowired
+    private LaunchRepository launchRepository;
+
+    @Autowired
+    private TestItemRepository testItemRepository;
+
+    @Autowired
+    private TestItemService testItemService;
+
+    @Autowired
+    private DataStoreService dataStoreService;
+
+    @Autowired
+    private CreateAttachmentHandler createAttachmentHandler;
+
+
+    private CreateLogHelper createLogHelper = this.new CreateLogHelper();
```

Comment:
any reason to use private class instead of bunch of private methods?

Reply:
no specific. changed it